rpmalloc/rpmalloc.c

```diff
@@ -702,6 +702,8 @@ static pthread_key_t _memory_thread_heap;
 #    endif
 #    if !defined(__clang__) && defined(__GNUC__)
 #      define _Thread_local __thread
+#    elif defined(__TINYC__)
+#      define _Thread_local
```

Comment:
This won't work, unless strictly for a single thread purpose. The allocator needs to have a per-thread heap pointer, if the compiler can't provide it out of the box it has to go through the pthread_key_t path.